### ISSUE TITLE
bug: sticky .selection no longer works in dev version

### ISSUE THREAD

**Issue by gofford** [glin/reactable#118]
reference: https://github.com/glin/reactable/blob/master/vignettes/sticky-columns-selection.Rmd

The example above shows how to make the selection column (`.selection`) sticky with horizontal scrolling.  This functionality no longer works in the latest dev version with identical code. Perhaps this is related the fix for #71. The issue is not present in v0.2.3 (CRAN).

reprex

``` r
library(shiny)
library(reactable)

ui <- fluidPage(
    mainPanel(
        reactableOutput("table")
    )
)

# Define server logic required to draw a histogram
server <- function(input, output) {
    output$table <- renderReactable({
        reactable(
            MASS::Cars93,
            selection = "multiple",
            columns = list(
                .selection = colDef(
                    style = list(position = "sticky", left = 0, background = "#fff", zIndex = 1),
                    headerStyle = list(position = "sticky", left = 0, background = "#fff", zIndex = 1),
                    width = 36
                ),
                Manufacturer = colDef(
                    style = list(position = "sticky", left = "36px", background = "#fff", zIndex = 1),
                    headerStyle = list(position = "sticky", left = "36px", background = "#fff", zIndex = 1)
                )
            ),
            onClick = "select",
            wrap = FALSE
        )
    })
}

# Run the application 
shinyApp(ui = ui, server = server)
```

<sup>Created on 2021-01-23 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.0 (2019-04-26)
#>  os       Debian GNU/Linux 9 (stretch)
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       Etc/UTC                     
#>  date     2021-01-23                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                         
#>  assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.6.0)                 
#>  callr         3.5.1      2020-10-13 [1] CRAN (R 3.6.0)                 
#>  cli           2.2.0      2020-11-20 [1] CRAN (R 3.6.0)                 
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.6.0)                 
#>  desc          1.2.0      2018-05-01 [1] CRAN (R 3.6.0)                 
#>  devtools      2.3.2      2020-09-18 [1] CRAN (R 3.6.0)                 
#>  digest        0.6.27     2020-10-24 [1] CRAN (R 3.6.0)                 
#>  ellipsis      0.3.1      2020-05-15 [1] CRAN (R 3.6.0)                 
#>  evaluate      0.14       2019-05-28 [1] CRAN (R 3.6.0)                 
#>  fansi         0.4.1      2020-01-08 [1] CRAN (R 3.6.0)                 
#>  fastmap       1.0.1      2019-10-08 [1] CRAN (R 3.6.0)                 
#>  fs            1.5.0      2020-07-31 [1] CRAN (R 3.6.0)                 
#>  glue          1.4.2      2020-08-27 [1] CRAN (R 3.6.0)                 
#>  highr         0.8        2019-03-20 [1] CRAN (R 3.6.0)                 
#>  htmltools     0.5.0      2020-06-16 [1] CRAN (R 3.6.0)                 
#>  htmlwidgets   1.5.2      2020-10-03 [1] CRAN (R 3.6.0)                 
#>  httpuv        1.5.4      2020-06-06 [1] CRAN (R 3.6.0)                 
#>  knitr         1.30       2020-09-22 [1] CRAN (R 3.6.0)                 
#>  later         1.1.0.1    2020-06-05 [1] CRAN (R 3.6.0)                 
#>  magrittr      2.0.1      2020-11-17 [1] CRAN (R 3.6.0)                 
#>  memoise       1.1.0      2017-04-21 [1] CRAN (R 3.6.0)                 
#>  mime          0.9        2020-02-04 [1] CRAN (R 3.6.0)                 
#>  pkgbuild      1.1.0      2020-07-13 [1] CRAN (R 3.6.0)                 
#>  pkgload       1.1.0      2020-05-29 [1] CRAN (R 3.6.0)                 
#>  prettyunits   1.1.1      2020-01-24 [1] CRAN (R 3.6.0)                 
#>  processx      3.4.5      2020-11-30 [1] CRAN (R 3.6.0)                 
#>  promises      1.1.1      2020-06-09 [1] CRAN (R 3.6.0)                 
#>  ps            1.5.0      2020-12-05 [1] CRAN (R 3.6.0)                 
#>  R6            2.5.0      2020-10-28 [1] CRAN (R 3.6.0)                 
#>  Rcpp          1.0.5      2020-07-06 [1] CRAN (R 3.6.0)                 
#>  reactable   * 0.2.3.9000 2021-01-23 [1] Github (glin/reactable@52b3adb)
#>  reactR        0.4.3      2020-07-12 [1] CRAN (R 3.6.0)                 
#>  remotes       2.2.0      2020-07-21 [1] CRAN (R 3.6.0)                 
#>  rlang         0.4.9      2020-11-26 [1] CRAN (R 3.6.0)                 
#>  rmarkdown     2.5        2020-10-21 [1] CRAN (R 3.6.0)                 
#>  rprojroot     2.0.2      2020-11-15 [1] CRAN (R 3.6.0)                 
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.6.0)                 
#>  shiny       * 1.5.0      2020-06-23 [1] CRAN (R 3.6.0)                 
#>  stringi       1.5.3      2020-09-09 [1] CRAN (R 3.6.0)                 
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.6.0)                 
#>  testthat      3.0.0      2020-10-31 [1] CRAN (R 3.6.0)                 
#>  usethis       1.6.3      2020-09-17 [1] CRAN (R 3.6.0)                 
#>  webshot       0.5.2      2019-11-22 [1] CRAN (R 3.6.0)                 
#>  withr         2.3.0      2020-09-22 [1] CRAN (R 3.6.0)                 
#>  xfun          0.19       2020-10-30 [1] CRAN (R 3.6.0)                 
#>  xtable        1.8-4      2019-04-21 [1] CRAN (R 3.6.0)                 
#>  yaml          2.2.1      2020-02-01 [1] CRAN (R 3.6.0)                 
#> 
#> [1] /usr/local/lib/R/site-library
#> [2] /usr/local/lib/R/library
```

</details>

**Comment by gofford** [comment]
Related to #71, in the scenario where both the selection box and the groupBy column are stickied also has a visual artefact where a blank portion passes in front of the groupBy column.

Test this by setting `groupBy = 'Manufacturer'` in the above example

**Comment by glin** [comment]
I can't reproduce this with that example, even after adding `groupBy = 'Manufacturer'`, but I think I know what you mean. The sticky column CSS is dependent on column position, so tables with sticky selection and groupBy columns will break because the column placement was changed in #71.

If this is the case, you'll have to flip the `left` positions in your column styles. Since the selection column is now the left-most column, it'll be the column with `left: 0` style now.

In reactable 0.2.3 on CRAN, sticky selection and groupBy columns would've been styled like this:
```r
library(reactable)

reactable(
  MASS::Cars93,
  selection = "multiple",
  columns = list(
    # groupBy column is the left-most column (left: 0)
    Manufacturer = colDef(
      style = list(position = "sticky", left = 0, background = "#fff", zIndex = 1),
      headerStyle = list(position = "sticky", left = 0, background = "#fff", zIndex = 1)
    ),
    # Selection column is the 2nd left column (left: 100 since columns have 100px width by default)
    .selection = colDef(
      style = list(position = "sticky", left = 100, background = "#fff", zIndex = 1),
      headerStyle = list(position = "sticky", left = 100, background = "#fff", zIndex = 1),
      width = 36
    )
  ),
  onClick = "select",
  wrap = FALSE,
  groupBy = "Manufacturer"
)
```

In the next version of reactable, it'll have to be styled like this:
```r
library(reactable)

reactable(
  MASS::Cars93,
  selection = "multiple",
  columns = list(
    # Selection column is the left-most column (left: 0)
    .selection = colDef(
      style = list(position = "sticky", left = 0, background = "#fff", zIndex = 1),
      headerStyle = list(position = "sticky", left = 0, background = "#fff", zIndex = 1),
      width = 36
    ),
    # groupBy column is the 2nd left column (left: 36 since we made the selection column 36px wide)
    Manufacturer = colDef(
      style = list(position = "sticky", left = 36, background = "#fff", zIndex = 1),
      headerStyle = list(position = "sticky", left = 36, background = "#fff", zIndex = 1)
    )
  ),
  onClick = "select",
  wrap = FALSE,
  groupBy = "Manufacturer"
)
```

There's not really a bug to fix here, but I'll mark #71 as a breaking change in NEWS for anyone that relied on the previous groupBy/selection column placement.

**Comment by gofford** [comment]
OK -- thank you for looking into it. I'll double check and see if I can get things working with that new info 👍
